# Analysis of the diversity of *Vanilla sp* (Orchidaceae)
The present repository contains a way to process and analyze data GBS, in order to identify the genetic diversity of individuals wild and cultivated in species of *Vanilla sp* genre. The GBS data was demultiplexed (Stacks) and aligned in front a reference genome given the nature of data. Later, the data was newly filtered (Stacks, VCFtools and R) to find Single Nucleotide Polymorphism (SNP) data in order to use to identify diversity among populations (*V. planifolia* wild and cultivated, *V. planifolia* between states of Mexico, and among species)

#### Prerequisites

**Software**

- [Stacks v.2.55](https://catchenlab.life.illinois.edu/stacks/) or superior
- [Trim Galore v.0.6.4](https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md) 
- [BWA (Burrows-Wheeler Aligner v.0.7.17)](http://bio-bwa.sourceforge.net/)
- [PicardTools v1.13](https://broadinstitute.github.io/picard/)
- [SamTools v1.10](http://www.htslib.org/)
- [Qualimap v2.2.1](http://qualimap.conesalab.org/)
- [VCFTools v 0.1.16](http://vcftools.sourceforge.net/)
- [R v 4.1.0 Camp Pontanezen](https://www.r-project.org/) y [Rstudio](https://www.rstudio.com/)

**R packages**

- SNPRelate (available from [Bioconductor project](https://www.bioconductor.org/packages/release/bioc/html/SNPRelate.html))
- Tidyverse
  - ggplot2

**PC info used in this proyect** 

- Distro: Linux Mint 20.1 Cinnamon v. 4.8.6
- Motherboard: Acer IPISB-AG
- Processor: Intel© Core™ i3-3240 CPU @ 3.40GHz × 2
- RAM: 8gb DDR3

#### Directories

**bin** 

Contains

- For_analysis_chloroplast
  1. 

#### Notes



#### Credits



Adrián Gutiérrez
